### PR TITLE
Fix Demon DLL

### PR DESCRIPTION
**Pull request by voidvxvi**
**Overview**
This PR fixes the Demon DLL crash caused by a null pointer dereference.
TLDR: `rundll32 demon.x64.dll,Start` will work again.

**Details**
`LdrModulePeb` is called in `DllMain` before `Instance` has been initialized.
https://github.com/HavocFramework/Havoc/blob/69ce17cb3686641e92a657cece5989feb9af64ed/payloads/Demon/src/main/MainDll.c#L47
`LdrModulePeb` accesses the uninitialized `Instance`, which is a null pointer derefernce at this point of execution, causing the process to crash.
https://github.com/HavocFramework/Havoc/blob/69ce17cb3686641e92a657cece5989feb9af64ed/payloads/Demon/src/core/Win32.c#L74

I’ve tested the x64 EXE, DLL, and SHC on Windows 11, with `WaitForSingleObjectEx`, `Foliage`, `Ekko` and `Zilean` sleep obfuscation enabled and all binaries are operating well with this update.

### CODE REVIEW

**Comment by M4rdc0re** [comment]
I tested it and the shellcode option with sleep jmp gadget "jmp rbx" is not working with your solution don't delete the information of "payloads/Demon/src/Demon.c" and it will work correctly. Actually, you only need to modify the file "payloads/Demon/src/main/MainDll.c"
